### DISCUSSION TITLE
can some people here please tell me how to enable the pluto tv plugin?

### DISCUSSION THREAD

**Discussion by merrychrishna** [streamlink/streamlink#4699] [Q&A] [answered]
https://github.com/streamlink/streamlink/blob/master/src/streamlink/plugins/pluto.py

can some people here please tell me how to enable the pluto tv plugin?
so far i have created a plugin directory in %appdata%streamlink/plugins.
then i placed the pluto.py from the streamlink github from the link i posted. i think now i just need to enable the plugin but i dont know how to do that exactly. there must be a command that i can put in the streamlink config file to enable it.

**Comment by bastimeyer** [comment, marked as answer]
https://github.com/streamlink/streamlink/discussions/4697#discussioncomment-3303793

> If you don't know how to actually use Streamlink, then I suggest you take a look at the documentation.
> https://streamlink.github.io/cli.html
> https://streamlink.github.io/cli/tutorial.html

**Read the documentation.**

You don't have to enable any built-in plugins. You are supposed to use the **canonical stream URL** from your web browser as input URL, and Streamlink will see that the Pluto plugin can handle this URL and it will then resolve the stream URL and output the stream, according to your other command-line parameters.

-----

**Example**

```sh
$ streamlink -l debug 'https://pluto.tv/en/live-tv/pluto-tv-star-trek-de' best
[cli][debug] OS:         Linux-5.18.14-1-git-x86_64-with-glibc2.35
[cli][debug] Python:     3.10.5
[cli][debug] Streamlink: 4.2.0+32.g9ddfa2f0.dirty
[cli][debug] Dependencies:
[cli][debug]  isodate: 0.6.1
[cli][debug]  lxml: 4.9.0
[cli][debug]  pycountry: 22.3.5
[cli][debug]  pycryptodome: 3.14.1
[cli][debug]  PySocks: 1.7.1
[cli][debug]  requests: 2.28.1
[cli][debug]  websocket-client: 1.3.2
[cli][debug] Arguments:
[cli][debug]  url=https://pluto.tv/en/live-tv/pluto-tv-star-trek-de
[cli][debug]  stream=['best']
[cli][debug]  --loglevel=debug
[cli][debug]  --player=mpv
[cli][info] Found matching plugin pluto for URL https://pluto.tv/en/live-tv/pluto-tv-star-trek-de
[plugins.pluto][debug] slug=pluto-tv-star-trek-de
[plugins.pluto][debug] app_version=6.4.0-3bcdc583b1e5a19f7836864e7ebe5e04d5e576c9
[plugins.pluto][debug] path=/stitch/hls/channel/6152ee71bf99590007893a11/master.m3u8
[utils.l10n][debug] Language code: en_US
[cli][info] Available streams: 570k (worst), 1000k, 1500k, 2100k, 3100k (best)
[cli][info] Opening stream: 3100k (hls-pluto)
[cli][info] Starting player: mpv
[stream.hls][debug] Reloading playlist
[cli][debug] Pre-buffering 8192 bytes
[stream.hls][debug] Segments in this playlist are encrypted
[stream.hls][debug] First Sequence: 18; Last Sequence: 22
[stream.hls][debug] Start offset: 0; Duration: None; Start Sequence: 20; End Sequence: None
[stream.hls][debug] Adding segment 20 to queue
[stream.hls][debug] Adding segment 21 to queue
[stream.hls][debug] Adding segment 22 to queue
[stream.hls][debug] Segment 20 complete
[cli.output][debug] Opening subprocess: mpv --force-media-title=https://pluto.tv/en/live-tv/pluto-tv-star-trek-de -
[stream.hls][debug] Segment 21 complete
[stream.hls][debug] Segment 22 complete
[cli][debug] Writing stream to output
[stream.hls][debug] Reloading playlist
[stream.hls][debug] Segments in this playlist are encrypted
[stream.hls][debug] Adding segment 23 to queue
[stream.hls][debug] Adding segment 24 to queue
[stream.hls][debug] Segment 23 complete
[cli][info] Player closed
[stream.segmented][debug] Closing worker thread
[stream.segmented][debug] Closing writer thread
[stream.hls][debug] Segment 24 complete
[cli][info] Stream ended
[cli][info] Closing currently open stream...
```

**Comment by merrychrishna** [comment]
i have to be doing something wrong because your example isnt working
![image](https://user-images.githubusercontent.com/59185469/182287544-62ddf679-7bd0-49a2-a4c2-46bc74eacc13.png)

**Comment by merrychrishna** [comment]
i also, tried removing the pluto.py file that i put in the plugins folder and get this
![image](https://user-images.githubusercontent.com/59185469/182288296-d12e60f2-19a4-4659-bc92-51fe02fca53a.png)

**Reply by bastimeyer** [reply]
First, you are using an outdated release (3.2.0 instead of 4.2.0), and second, your install is probably modified/broken, considering your previous comments. Do a full uninstall and a clean re-install.

The plugin is working fine, as you can see from the output I posted above, and if Streamlink can't find a plugin for this specific input URL, then you are indeed doing something wrong with your setup.

**Reply by merrychrishna** [reply]
oh ok. ill uninstall and reinstall right now. thanks

**Comment by merrychrishna** [comment]
i think i found the problem. my streamlink was 3.2.0 x86 so when i updated streamlink the path was wrong. i have now fixed the path and see the updated streamlink but i have a new rpoblem. i get this error.
![image](https://user-images.githubusercontent.com/59185469/182289012-4fdf0e0c-af25-40a8-934d-d73c4a2571ce.png)

**Comment by merrychrishna** [comment]
i have now uninstalled all versions of streamlink via control panel>programs and featuress. then i rebooted. then i typed streamlink in the cmd prompt but one last version of streamlink was still being called which was version 4.1.0 if i remember correctly. i did a "where streamlink" that lead me to streamlink exe in a python 37 folder so i deleted the streamlink exe. then i installed "streamlink 4.2.0 x86 x64 310.exe. but still i get this error.
![image](https://user-images.githubusercontent.com/59185469/182291600-b0ec5afa-8efe-445c-a18a-3599ee461e40.png)

**Reply by bastimeyer** [reply]
This looks like a bug in the Windows installer's build config in regards to the `--loglevel=debug` parameter (`-l debug`). Just don't set the parameter, so you don't trigger the code path which raises the error.

You can also downgrade to 4.1.0, which doesn't include the more detailed dependency version list added in #4575.

----

Looks like the `dist-info` of the streamlink package is missing from the Windows installer's content, so `importlib.metadata` can't find the dependencies of the streamlink package. I'll have a look at this later.

**Reply by bastimeyer** [reply]
> `importlib.metadata.PackageNotFoundError: No package metadata was found for streamlink`

This is fixed now with the latest release of Streamlink's Windows installer, `4.2.0-2`:
- https://github.com/streamlink/windows-builds/releases/tag/4.2.0-2
- https://github.com/streamlink/windows-builds/blob/master/CHANGELOG.md#420-2-2022-08-02
- https://github.com/streamlink/windows-builds/commit/b177682ab0295e68f0e9ffe6623f9a68dfdec95e

**Comment by merrychrishna** [comment]
i have now uninstalled 4.2.0-1 and have installed 4.2.0-2. theres no longer an error but the plugin is not seeing the url.
![image](https://user-images.githubusercontent.com/59185469/182406145-ffeac609-fbd3-496b-a45f-e2061da221ad.png)

**Comment by merrychrishna** [comment]
if i replace your single quotes with double quotes i get this
![image](https://user-images.githubusercontent.com/59185469/182406961-b7982f55-9d8c-4d45-a925-30661706fcb3.png)

**Reply by mkbloke** [reply]
It won't find that URL unless you're in Germany.  If you visit that URL in a web browser from a non-German IP address it will likely redirect to other content on the Pluto platform.  Try another URL for content that you know you can access via a web browser.

**Reply by merrychrishna** [reply]
oh i just realised that so i tried streamlink -l debug "https://pluto.tv/live-tv/80s-rewind" best and its working now
